src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -2722,7 +2723,7 @@ LogicalResult ONNXDequantizeLinearOp::inferShapes(
     std::function<void(mlir::Region &)> doShapeInference) {
   auto inTy = x().getType().dyn_cast<RankedTensorType>();
   if (!inTy || !inTy.hasStaticShape()) {
-    return emitOpError("Input is not a statically-shaped type");
+    return success();
```

Comment:
Question: Some emit an error if not ranked, other if not statically shaped, like here. How do you keep the distinction later on?

Reply:
It's my fault that I should separate them here. In long term, we may define a  trait for op that requires special handling.

Reply:
@chentong319 so what is the proposed solution to address this. If it is not done in this PR, we will loose these conditions and it will be much harder to add them later.

Reply:
I checked the code and found the extra condition is not really needed. I'm deleting them now.

Reply:
@chentong319 so you went through all conditions that are not "must have rank" and none of these conditions are needed?